pkg/server/plugin/zmq/rpc.go

```diff
@@ -176,6 +194,54 @@ func (p *zmqTransport) ipc(req *pluginrequest.Request) (out []byte, err error) {
 	return
 }
 
+func (p *zmqTransport) handleRequest(parcel *Parcel) {
+	workerID := parcel.Worker
+	requestID := parcel.RequestID
+	bounceCount := parcel.BounceCount
+	buffer := parcel.Frame
+	responseChan := parcel.RespChan
+
+	reader := bytes.NewReader(buffer)
+	data := map[string]interface{}{}
+	if jsonErr := json.NewDecoder(reader).Decode(&data); jsonErr != nil && jsonErr != io.EOF {
+		// TODO(b123400): handle json error
+		return
+	}
+
+	payloadData := data["payload"].(map[string]interface{})
+	method := data["method"].(string)
+	path := data["path"].(string)
+
+	payload := &router.Payload{
+		Context: context.Background(),
+		Meta: map[string]interface{}{},
+		Data: payloadData,
+		AccessKey: router.MasterAccessKey,
+	}
+
+	responseWriter := &zmqResponseWriter{}
+	resp := router.NewResponse(responseWriter)
+
+	newCtx := context.WithValue(payload.Context, ZMQWorkerIDContextKey, workerID)
+	newCtx = context.WithValue(newCtx, ZMQRequestIDContextKey, requestID)
+	newCtx = context.WithValue(newCtx, ZMQBounceCountContextKey, bounceCount)
+	payload.Context = newCtx
+
+	// TODO(b123400), put real path and method here
+	p.router.HandlePayload(path, method, payload, resp)
+
+	responseChan <- responseWriter.response
+}
+
+func (p *zmqTransport) listenRequests() {
+	for {
+		select {
+		case parcel := <- p.broker.ReqChan:
+			go p.handleRequest(parcel)
```

Comment:
Okay I probably have a different picture from yours. In my mind Broker is a mediator between skygear-server (i.e. zmq transport currently) and zmq workers. So the primary role of broker is to exchange messages between transport and workers. Previously only transport knows about broker because requests are only initiated from skygear-server side. Now we allow workers to initiate requests. And it appear naturally to me for broker to know about both ends and exchange messages between them. I cannot speak for whether others in the team have this same picture as mine though.